source/exe/hot_restart.cc

```diff
@@ -307,7 +308,17 @@ void HotRestartImpl::sendMessage(sockaddr_un& address, RpcBase& rpc) {
 
 void HotRestartImpl::onGetListenSocket(RpcGetListenSocketRequest& rpc) {
   RpcGetListenSocketReply reply;
-  reply.fd_ = server_->getListenSocketFd(std::string(rpc.address_));
+  reply.fd_ = -1;
+
+  Network::Address::InstanceConstSharedPtr addr =
+      Network::Utility::resolveUrl(std::string(rpc.address_));
+  for (const auto& listener : server_->listenerManager().listeners()) {
+    if (listener.get().socket().localAddress()->asString() == addr->asString()) {
+      reply.fd_ = listener.get().socket().fd();
```

Comment:
This logic is only used here and can be implemented on top of the existing interface. It seemed unnecessary to add an interface function for just this callsite. Happy to do it though if you think useful.